compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -1093,28 +1093,27 @@ fn try_optimize_array_get_from_previous_set(
     // Arbitrary number of maximum tries just to prevent this optimization from taking too long.
     let max_tries = 5;
     for _ in 0..max_tries {
-        match &dfg[array_id] {
-            Value::Instruction { instruction, .. } => {
-                match &dfg[*instruction] {
-                    Instruction::ArraySet { array, index, value, .. } => {
-                        if let Some(constant) = dfg.get_numeric_constant(*index) {
-                            if constant == target_index {
-                                return SimplifyResult::SimplifiedTo(*value);
-                            }
-
-                            array_id = *array; // recur
-                        } else {
-                            return SimplifyResult::None;
+        if let Some(instruction) = dfg.get_instruction(array_id) {
```

Comment:
Do we expect array_set to be used on global arrays at all? I'd expect/hope this wouldn't be needed here

Reply:
It is needed for the `MakeArray` case not the `ArraySet` case. I could assert inside of the DFG method that the instruction must be `MakeArray` for safety.

Reply:
I added an insertion inside of the dfg method.